embassy-stm32/src/usart/mod.rs

```diff
@@ -136,6 +136,40 @@ pub enum HalfDuplexReadback {
     Readback,
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+/// Half duplex IO mode
+pub enum OutputConfig {
+    /// Push pull allows for faster baudrates, no internal pullup
+    PushPull,
+    #[cfg(not(gpio_v1))]
+    /// Push pull with internal pull up resistor
+    PushPullPullUp,
+    #[cfg(not(gpio_v1))]
+    /// Push pull with internal pull down resistor
+    PushPullPullDown,
+    /// Open drain output using external pull down resistor
+    OpenDrainExternal,
+    #[cfg(not(gpio_v1))]
+    /// Open drain output using internal pull up resistor
+    OpenDrainInternal,
```

Comment:
I'd call these OpenDrain and OpenDrainPullUp. This is more consistent with the GPIO driver. It doesn't concern what's on the board, just what the MCU itself is doing (enable pullup, or not enable pullup)